10.1/root/usr/share/container-scripts/mysql/common.sh

```diff
@@ -162,3 +165,63 @@ function wait_for_mysql_master() {
     sleep 1
   done
 }
+
+# Converts string version to the integer format (5.5 is converted to 505,
+# 10.1 is converted into 1001, etc.
+function version2number() {
+  local version=0
+  local version_text="$1"
+  local version_major=$(echo "$version_text" | sed -e 's/\([0-9]*\)\.\([0-9]*\)\..*$/\1/')
+  local version_minor=$(echo "$version_text" | sed -e 's/\([0-9]*\)\.\([0-9]*\)\..*$/\2/')
+  if [[ $version_major =~ ^[0-9]+$ ]] && [[ $version_minor =~ ^[0-9]+$ ]] ; then
+    version=$((version_major*100+version_minor))
+  fi
+  if [ "$version" -eq 0 ] ; then
+    echo "Error: version '$version_text' cannot be converted to the number."
+    echo "0"
+  fi
+  echo "${version}"
+}
+
+# Prints version of the mysqld that is currently available (string)
+function mysqld_version() {
+  ${MYSQL_PREFIX}/libexec/mysqld -V | awk '{print $3}'
+}
+
+# Returns version from the daemon in integer format
+function mysqld_compat_version() {
+  local compat_version=$(mysqld_version)
+  echo $(version2number $(mysqld_version))
+}
+
+# Returns version from the datadir
+function get_datadir_version() {
+  local datadir="$1"
+  local upgrade_info_file=$(get_mysql_upgrade_info_file "$datadir")
+  local version_text=$(cat "$upgrade_info_file" | head -n 1)
+  local version=$(version2number "${version_text}")
+
+  # version of the data could not be determined nor guessed
+  if [ "$version" -eq 0 ] ; then
+    local mysqld_version=$(mysqld_version)
+    echo "Warning: Version of the data could not be determined. Running such a container is very risky. The current daemon version is ${mysqld_version}. If you are sure that the data dir format corresponds with this version, create a file called 'mysql_upgrade_info' that includes just the string '${mysqld_version}' in the root of the data directory or run 'mysql_upgrade' utility that creates such a file as well. That will enable correct upgrade check in the future."
+    return 1
+  fi
+
+  echo "${version}"
+}
+
+# Returns name of the file in the datadir that holds version information about the data
+function get_mysql_upgrade_info_file() {
+  local datadir="$1"
+  echo "$datadir/mysql_upgrade_info"
+}
+
+# Gets version of the current daemon and stores it into the appropriate file in the datadir
+function store_mysqld_version() {
+  local datadir="$1"
+  local version=$(mysqld_version)
+  local upgrade_info_file=$(get_mysql_upgrade_info_file "$datadir")
+  echo "${version}" > "${upgrade_info_file}"
+  log_info "Storing version '${version}' information into the data dir '${upgrade_info_file}'"
```

Comment:
It's replaced by mysql_upgrade as described above, so I'll probably remove this function at all..